### DISCUSSION TITLE
상호작용 애니메이션을 캐릭터가 아니고 오브젝트에 넣고 싶습니다.

### DISCUSSION THREAD

**Discussion by sul321** [naverz/zepeto-studio-kor#1243] [5. Q&A]
안녕하세요 
아무것도 모르는 그냥 가이드만 보고 작업하는 왕초보입니다...
상호작용을 만들고자 하는데  무작정 따라하기 씬3을 참고하여 플레이어가 움직이는 건 변경하여 사용할 수 있겠는데요
상자나 오브젝트에 애니메이션을 적용하여 움직이게 하는 스크립트 예제를 혹시 알려 주실 수 있을까요? 

import { Canvas, AnimationClip, WaitForSeconds, GameObject, Object, Random } from 'UnityEngine';
import { Button } from 'UnityEngine.UI';
import { ZepetoPlayers, ZepetoCharacter } from 'ZEPETO.Character.Controller';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script';

// import custom script from path
import UIController from './UIController';

export default class Interaction extends ZepetoScriptBehaviour 
{
    public openUIGesture: Button;
    public interactionCanvas: Canvas;
    public animationClip: AnimationClip;
    public uiControllerObject: GameObject;
    public badEffectFactory: GameObject;
    public goodEffectFactory: GameObject;
    public gift: GameObject;
    public failureRatio: number = 50;
    
    private uiController: UIController;
    private zepetoCharacter :ZepetoCharacter;

    Start() 
    {    
        // Set EventCamera 
        this.interactionCanvas.worldCamera = ZepetoPlayers.instance.ZepetoCamera.camera;
        // Set character
        ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
            this.zepetoCharacter = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character;
        });
        // Script import
        this.uiController = this.uiControllerObject.GetComponent<UIController>();
        
        //Button Hide
        this.openUIGesture.gameObject.SetActive(false);
        
        //When Button Click
        this.openUIGesture.onClick.AddListener(()=>{
            this.zepetoCharacter.SetGesture(this.animationClip);
            this.StartCoroutine(this.FirstRoutine());         
        });
    }
    
    OnTriggerEnter(collider)
    {
        this.openUIGesture.gameObject.SetActive(true);
    }
    OnTriggerExit(collider)
    {
        this.openUIGesture.gameObject.SetActive(false);
    }
    
    *FirstRoutine() 
    {
        this.uiController.Loading();
        yield new WaitForSeconds(3);
        this.zepetoCharacter.CancelGesture();
        this.RandomCalculation();
    }

    private RandomCalculation()
    {
        let randomNumber: number;
        randomNumber = Random.Range(0,100);
        if (randomNumber <= this.failureRatio)
        {
            this.Lose();
        }
        else
        {
            this.Win();
        }
        this.StartCoroutine(this.SecondRoutine());
    }

    private Lose()
    {
        this.uiController.Lose();
        //GameObject Create
        var obj = Object.Instantiate(this.badEffectFactory) as GameObject;
        obj.transform.position = this.transform.position;
    }

    private Win()
    {
        this.uiController.Win();
        //GameObject Create
        var obj = Object.Instantiate(this.goodEffectFactory) as GameObject;
        obj.transform.position = this.transform.position;
        var giftobj = Object.Instantiate(this.gift) as GameObject;
        giftobj.transform.position = this.transform.position;
    }

    *SecondRoutine()
    {
        yield new WaitForSeconds(1);
        this.uiController.Init();
        //Destroy box
        GameObject.Destroy(this.gameObject);
    }
}

여기서 zepetoCharacter 부분만 GameObject로 변경 하면 될까요?

**Comment by bestdeveloperjo** [comment]
오브젝트와 캐릭터가  상호 작용 할려면 trigger를 쓰면 좋을거 같습니다.
https://funfunhanblog.tistory.com/13 

애니메이션 관련 
https://www.youtube.com/watch?v=0am9N6xFGxk&t=643s 해당 영상 참고 하시면 좋을거 같습니다
